analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -363,14 +364,44 @@ abstract class PackageManager(
 }
 
 /**
- * Parse a string with metadata about an [author] to extract the author name. Many package managers support
- * such author information in string form that contain additional properties like an email address or a
- * homepage. These additional properties are typically separated from the author name by specific [delimiters],
- * e.g. the email address is often surrounded by angle brackets. This function assumes that the author name is the
- * first portion in the given [author] string before one of the given [delimiters] is found.
+ * Parse a string with metadata about an [author] that several package managers use and try to extract the author's
+ * name, email address, and homepage. These properties are typically surrounded by specific delimiters, e.g. the email
+ * address is often surrounded by angle brackets (see [emailDelimiters]) and the homepage is often surrounded by
+ * parentheses (see [homepageDelimiters]). Return [AuthorInfo] for these properties where unavailable ones are set to
+ * null.
  */
-fun parseAuthorString(author: String?, vararg delimiters: Char = charArrayOf('<')): String? =
-    author?.split(*delimiters, limit = 2)?.firstOrNull()?.trim()?.ifEmpty { null }
+fun parseAuthorString(
+    author: String?,
+    emailDelimiters: Pair<Char, Char> = '<' to '>',
+    homepageDelimiters: Pair<Char, Char> = '(' to ')'
+): AuthorInfo {
+    if (author == null) return AuthorInfo(null, null, null)
+
+    var cleanedAuthor = author
+    var email: String? = null
+    var homepage: String? = null
+
+    val e = emailDelimiters.toList().map { Regex.escape(it.toString()) }
```

Comment:
Is it necessary to have the delimiters as parameters? Otherwise, we could avoid constructing regexes each run.

Reply:
I didn't want to remove this configurability as well during the refactoring. But I agree that it could probably be hard-coded as a follow-up.

Reply:
ok, let's just keep it for now.